### PR TITLE
test: scripts to build/deploy on OpenShift

### PR DESCRIPTION
**Pull request by xcoulon**
a single script with 2 funcs to:
1. expose the Container Registry with a
  public route: 
```
$ ./scripts/deploy-dev.sh setup
```
2. build/push the Docker image, then patch
  the deployment: 
```
$ ./scripts/deploy-dev.sh refresh
```

also: changed the Dockerfile to fasten the builds
(fewer layers to change after the binary was copied)

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: scripts/redeploy-app.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# copy/replace the binary into the pod's container
+echo "🚚 pushing the image into the Container registry"
+podman push --tls-verify=false $REGISTRY_ROUTE/$HOST_NS/$IMAGE_NAME
+echo "✅ done"
+
+# restart the process in the pod's container
+INTERNAL_REGISTRY=image-registry.openshift-image-registry.svc:5000
+echo "✏️ patching the deployment with image $INTERNAL_REGISTRY/$HOST_NS/$IMAGE_NAME"
+oc patch deployment/registration-service --type='json' -p="[{\"op\": \"replace\", \"path\": \"/spec/template/spec/containers/0/image\", \"value\":\"$INTERNAL_REGISTRY/$HOST_NS/$IMAGE_NAME\"}]"
```
Comment (MatousJobanek) [inline scripts/redeploy-app.sh:37]: so it expects that you are in the context of the namespace where the registration-service is deployed, right?
Maybe worth putting it into the description at the very beginning of the script.

**Review by MatousJobanek — APPROVED** [review]
Cool :+1:

File: scripts/redeploy-app.sh (hunk 1 above)
Comment (xcoulon) [reply scripts/redeploy-app.sh:37]: good point!

**Review by alexeykazakov — COMMENTED** [review]
Could you also update readme and mention how to build and deploy the reg-service on OpenShift using this script and what exactly it does?

**Review by rajivnathan — APPROVED** [review]

**Comment by xcoulon** [comment]
> update README.adoc

done in https://github.com/codeready-toolchain/registration-service/pull/270/commits/d65964cae90694d67e4b0e7e5d8fc71bcc42289e

File: scripts/redeploy-app.sh (hunk 1 above)
Comment (xcoulon) [reply scripts/redeploy-app.sh]: see 4aeecb9

**Pull request merged**